src/Elastic.Apm/DiagnosticListeners/HttpDiagnosticListenerImplBase.cs

```diff
@@ -140,10 +145,9 @@ private void ProcessStartEvent(TRequest request, Uri requestUrl)
 			string HeaderGetter(string header) => RequestTryGetHeader(request, header, out var value) ? value : null;
 
 			ISpan span = null;
-
-			if (_realAgent != null)
+			if (_configuration != null)
 			{
-				using (var httpTracers = _realAgent.Components.GetHttpTracers())
+				using (var httpTracers = _configuration.GetTracers())
```

Comment:
This is not new in this commit, but now that I see this again I was wondering: this happens for every outgoing HTTP request and this allocates. Not a blocker, but it'd be nice to somehow avoid allocation here.

Reply:
There could be a check to see if there are any http tracers before allocating

Reply:
Added check